components/supervisor/pkg/serverapi/publicapi.go

```diff
@@ -259,57 +263,32 @@ func (s *Service) listenInstanceUpdate(ctx context.Context, instanceID string) {
-	getData := func() (*gitpod.WorkspaceInstance, error) {
-		if !s.usePublicAPI(ctx) {
-			return s.lastServerInstance, nil
-		}
-		service := v1.NewWorkspacesServiceClient(s.publicAPIConn)
-		resp, err := service.GetWorkspace(ctx, &v1.GetWorkspaceRequest{
-			WorkspaceId: workspaceID,
-		})
-		if err != nil {
-			log.WithField("method", "GetWorkspace").WithError(err).Error("failed to call PublicAPI")
-			return nil, err
-		}
-		instance := &gitpod.WorkspaceInstance{
-			CreationTime: resp.Result.Status.Instance.CreatedAt.String(),
-			ID:           resp.Result.Status.Instance.InstanceId,
-			Status: &gitpod.WorkspaceInstanceStatus{
-				ExposedPorts: []*gitpod.WorkspaceInstancePort{},
-				Message:      resp.Result.Status.Instance.Status.Message,
-				// OwnerToken:   "", not used so ignore
-				Phase:   resp.Result.Status.Instance.Status.Phase.String(),
-				Timeout: resp.Result.Status.Instance.Status.Conditions.Timeout,
-				Version: int(resp.Result.Status.Instance.Status.StatusVersion),
-			},
-			WorkspaceID: resp.Result.WorkspaceId,
-		}
-		for _, port := range resp.Result.Status.Instance.Status.Ports {
-			info := &gitpod.WorkspaceInstancePort{
-				Port: float64(port.Port),
-				URL:  port.Url,
-			}
-			if port.Policy == v1.PortPolicy_PORT_POLICY_PUBLIC {
-				info.Visibility = gitpod.PortVisibilityPublic
-			} else {
-				info.Visibility = gitpod.PortVisibilityPrivate
-			}
-			instance.Status.ExposedPorts = append(instance.Status.ExposedPorts, info)
-		}
-		return instance, nil
-	}
-	exp := &backoff.ExponentialBackOff{
-		InitialInterval:     2 * time.Second,
-		RandomizationFactor: 0.5,
-		Multiplier:          1.5,
-		MaxInterval:         30 * time.Second,
-		MaxElapsedTime:      0,
-		Stop:                backoff.Stop,
-		Clock:               backoff.SystemClock,
-	}
-	return backoff.RetryWithData(getData, exp)
-}
+// func (s *Service) getWorkspaceInfo(ctx context.Context, instanceID, workspaceID string) (*gitpod.WorkspaceInstance, error) {
+// 	getData := func() (*gitpod.WorkspaceInstance, error) {
+// 		if !s.usePublicAPI(ctx) {
+// 			return s.lastServerInstance, nil
+// 		}
+// 		service := v1.NewWorkspacesServiceClient(s.publicAPIConn)
+// 		resp, err := service.GetWorkspace(ctx, &v1.GetWorkspaceRequest{
+// 			WorkspaceId: workspaceID,
+// 		})
+// 		if err != nil {
+// 			log.WithField("method", "GetWorkspace").WithError(err).Error("failed to call PublicAPI")
+// 			return nil, err
+// 		}
+// 		return workspaceStatusToWorkspaceInstance(resp.Result.Status), nil
+// 	}
+// 	exp := &backoff.ExponentialBackOff{
+// 		InitialInterval:     2 * time.Second,
+// 		RandomizationFactor: 0.5,
+// 		Multiplier:          1.5,
+// 		MaxInterval:         30 * time.Second,
+// 		MaxElapsedTime:      0,
+// 		Stop:                backoff.Stop,
+// 		Clock:               backoff.SystemClock,
+// 	}
+// 	return backoff.RetryWithData(getData, exp)
+// }
```

Comment:
FWIW, I'd remove this. You always have it in git history, if you need it but commented out code only confuses people when they read it later on.